external-crates/move/crates/move-vm-profiler/src/lib.rs

```diff
@@ -319,7 +318,21 @@ macro_rules! profile_close_instr {
 #[macro_export]
 macro_rules! profile_dump_file {
     ($profiler:expr) => {
-        #[cfg(debug_assertions)]
+        #[cfg(feature = "gas-profiler")]
         $profiler.to_file()
     };
 }
+
+#[cfg(feature = "gas-profiler")]
+#[macro_export]
+macro_rules! gas_profiler_feature {
+    ($($tt:tt)*) => {
```

Comment:
Note: the `compile_error` I mentioned in the `sui-node` file may cause the `tt` macro type to not be happy. But I believe switching this to an `expr` here and below would resolve if that's the case.

Reply:
The issue I ran into was that the compile error is always thrown in CI, and that is because cargo always first builds a version with the union of all features, which is guaranteed to upset the compiler if we create such a compile error :(